File: workout_logger_app/lib/models/exercise.dart
class Exercise {
  final String id;
  final String routineId;
  final String name;
  final int totalSets;
  final String userId;

  Exercise({
    required this.id,
    required this.routineId,
    required this.name,
    required this.totalSets,
    required this.userId,
  });

  factory Exercise.fromJson(Map<String, dynamic> json) {
    return Exercise(
      id: json['_id'],
      routineId: json['routineId'],
      name: json['name'],
      totalSets: (json['totalSets'] ?? 0) as int,
      userId: json['userId'] ?? '',
    );
  }
}

File: workout-logger-backend/models/Exercise.js
const mongoose = require('mongoose');

const ExerciseSchema = new mongoose.Schema({
  routineId: { type: mongoose.Schema.Types.ObjectId, ref: 'Routine', required: true },
  name: { type: String, required: true },
    userId:    { type: String, required: true }, 
  totalSets:{ type: Number, required: true },
  createdAt: { type: Date, default: Date.now },
    order_position: { type: Number, default: 0 }, 
});
ExerciseSchema.index({ routineId: 1, name: 1 });
module.exports = mongoose.model('Exercise', ExerciseSchema);
